### DISCUSSION TITLE
Relay delay time off

### DISCUSSION THREAD

**Discussion by Surleika** [HomeSpan/HomeSpan#796] [Show and tell]
Hi,
I have a relay, witch opens the door by switching on and delayed switching off. Switch on is done for example by pushing the button in the app or using a nfc token to trigger a automation. 

The switch off will be delayed switching off after a specific time programmed in the esp32 homespan device itself. Afterwards the state should be sent to the Homekit App so that the tile can be updated. 

Any chance that there is a refering command to update the change of a pin state to homespan?

**Comment by HomeSpan** [comment]
This is readily doable in HomeSpan.  The `setVal()` method lets you update the value of a Characteristic in the Home App and the `timeVal()` method allows you to keep track of the time since a Characteristic has changed.  Using the two methods together allows you to turn on a relay for a set period of time and have the Home App stay updated.

See Tutorial Example [14-EmulatedPushButtons](https://github.com/HomeSpan/HomeSpan/tree/master/examples/14-EmulatedPushButtons) for a sketch that does what I think you are looking for.
